### DISCUSSION TITLE
Suggestion: Spies to know invasion point in advance

### DISCUSSION THREAD

**Discussion by ben-strasser** [Keriew/augustus#1053] [Ideas]
Some maps have many invasion points and shifting units between them can be difficult. This can be quite frustrating if the invasion comes from a different point than one expects. It would be nice to know in advance where an invasion hits and how large it will be. Just always telling the player where invasions come from, however, removes some fun that can comes from being surprised.

One possible solution would be to add a spy system mechanic to the game. This could be as simple as another building and if you have it, the game tells you a few months before the invasion, what the invasion point will be from where it comes.

There are a lot of possible variations. For example, one could require one spy building per invasion point and then the player is only told whether the invasion will come from the monitored invasion points or not. I could list more possible variations. However, I would first like to gather some feedback on whether this sounds like a good idea to anyone. :)

**Comment by Ouaz** [comment]
This has been discussed but no consensus emerged (some doesn't want it at all, while others are not agreed on how to implement it).

For my part, I always found awkward that the Advisor warns "the Enemy is at the door" of the city, but there's no way to know at least the global direction from where the invasion comes (N, S, E, W).

Worst sentries ever. :laughing:

**Reply by mpdmtr** [reply]
Personally I see the possible spy mechanics as following:
1. In a caravanserai one can conclude an "agreement of espionage" for some amount of gold/wine/other resource. This "agreement" consumes some resources per month (for example, 1 gold / month).
2. During action of the agreement player see all invasion points. Personally I think, it's enough to know them to take precautions (surround by walls, forts, etc.).
3. When gold in the storages ends, the agreement expires and invasion points disappear. 
4. Also it could be added a possibility simply pay the enemy army / Caesar's army some resource (gold/wine/something else) to prevent attack (like it's done in "Zeus") during action of the agreement. The resource could be randomized or defined in the editor by map creator.
